components/mod-event/useModEventList.tsx

```diff
@@ -321,3 +354,39 @@ export const useModEventList = (
     addToWorkspace,
   }
 }
+
+const TagBasedTypeFilters = {
+  [MOD_EVENTS.DISABLE_DMS]: { add: DM_DISABLE_TAG, remove: '' },
+  [MOD_EVENTS.ENABLE_DMS]: { remove: DM_DISABLE_TAG, add: '' },
+  [MOD_EVENTS.DISABLE_VIDEO_UPLOAD]: {
+    add: VIDEO_UPLOAD_DISABLE_TAG,
+    remove: '',
+  },
+  [MOD_EVENTS.ENABLE_VIDEO_UPLOAD]: {
+    remove: VIDEO_UPLOAD_DISABLE_TAG,
+    add: '',
+  },
+}
+
+const buildTagFilter = (type: string) => {
+  const add: string[] = []
+  const remove: string[] = []
+  const tagTypes = Object.keys(TagBasedTypeFilters)
+
+  if (!tagTypes.includes(type)) {
+    return { add, remove }
+  }
+
+  tagTypes.forEach((key) => {
+    if (type === key) {
+      if (TagBasedTypeFilters[key].add) {
+        add.push(TagBasedTypeFilters[key].add)
+      }
+      if (TagBasedTypeFilters[key].remove) {
+        remove.push(TagBasedTypeFilters[key].remove)
+      }
+    }
+  })
```

Comment:
Looks like that iteration is not really needed:

```suggestion
  if (!(type in tagTypes) || !Object.hasOwn(tagTypes, type)) {
    return { add, remove }
  }
  if (TagBasedTypeFilters[type].add) {
    add.push(TagBasedTypeFilters[type].add)
  }
  if (TagBasedTypeFilters[type].remove) {
    remove.push(TagBasedTypeFilters[type].remove)
  }
```

Reply:
nice thanks!